### PR TITLE
Fix global index deletion when projection type is INCLUDE

### LINKED ISSUE #1603: [BUG] Global indexes with projection attributes are deleted while updating the table.
### Summary:

When a table is configured to have GSI with custom projection attributes, then they are being deleted when table `update` is set to true.

### Problem

I did debug the issue. It is caused because of the invalid comparison of the index properties.

In the function `deleteIndexes` of `utils.dynamoose.index_changes.ts` the following condition fails to return false.

```
return !(expectedIndexes.GlobalSecondaryIndexes || []).find((searchIndex) => obj.equals(obj.pick(cleanedIndex, identicalProperties), obj.pick(searchIndex as any, identicalProperties)));
```

This is because the order of `cleanedIndex.Projection.NonKeyAttributes` is not same as `searchIndex.Projection.NonKeyAttributes` (they are differently ordered by AWS)

### Suggested solution:

Projected attributes should be sorted to pass the `equals` comparision.

#### Schema
```
const schema = new dynamoose.Schema(
    {
      id: {
        type: String,
        required: true,
        hashKey: true,
      },
      group: {
        type: String,
        required: true,
        index: {
            name: 'group-gsi',
            type: 'global',
            project: ['id', 'name', 'email', 'location'],
        }
     },
     name: String,
     email: String,
     location: String
);
```

#### Model
```
dynamoose.model('TableName', schema, {
  create: true,
  update: true,
});
```

### Current output and behavior (including stack trace):

the index `group-gsi` is deleted every time even though there are no changes to it.

### Expected output and behavior:

Index should not be deleted without changes

### Environment:

Operating System: MacOS
Operating System Version: 13.3.1
Node.js version (`node -v`): 18
NPM version: (`npm -v`): 9
Dynamoose version: 3.2.0

### Other information (if applicable):

NA

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

### PR DESCRIPTION
**Pull request by madhusudhand**
### Summary:

Fixes issue where index is deleted when projection is set to `list of attributes` instead of `ALL`.

#### Schema
```
const schema = new dynamoose.Schema(
    {
      id: {
        type: String,
        required: true,
        hashKey: true,
      },
      group: {
        type: String,
        required: true,
        index: {
            name: 'group-gsi',
            type: 'global',
            project: ['data1', 'data3', 'data2'],
        }
     },
     data1: String,
     data2: String,
     data3: String
);
```

#### Model
```
dynamoose.model('TableName', schema, {
  create: true,
  update: true,
});
```

### GitHub linked issue:

Closes #1603

### Type (select 1):
- [x] Bug fix
- [ ] Feature implementation
- [ ] Documentation improvement
- [ ] Testing improvement

- [ ] Test added to report bug (GitHub issue #---- @---)
- [ ] Something not listed here

### Is this a breaking change? (select 1):
- [ ] 🚨 YES 🚨
- [x] No
- [ ] I'm not sure

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Are all the tests currently passing on this PR? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have read through and followed the Contributing Guidelines
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [x] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoose/dynamoose/blob/main/LICENSE)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have filled out all fields above

### CODE REVIEW

**Comment by madhusudhand** [comment]
I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

**Comment by madhusudhand** [comment]
Hey can you reopen and review this?

**Comment by madhusudhand** [comment]
Hi, could you review and merge this change?
without this I cannot use indexes with specific projection list

**Comment by madhusudhand** [comment]
@fishcharlie can you consider merging this?

File: packages/dynamoose/lib/utils/dynamoose/index_changes.ts [hunk 1]
```diff
@@ -42,7 +49,10 @@ const index_changes = async (table: Table, existingIndexes = []): Promise<(Model
 			}
 		});
 
-		return !(expectedIndexes.GlobalSecondaryIndexes || []).find((searchIndex) => obj.equals(obj.pick(cleanedIndex, identicalProperties), obj.pick(searchIndex as any, identicalProperties)));
+		return !(expectedIndexes.GlobalSecondaryIndexes || []).find((searchIndex) => obj.equals(
+			sanitizeIndex(obj.pick(cleanedIndex, identicalProperties) as unknown as IndexItem), 
+			sanitizeIndex(obj.pick(searchIndex as any, identicalProperties) as unknown as IndexItem)
```
Comment (fishcharlie) [inline packages/dynamoose/lib/utils/dynamoose/index_changes.ts:54]:
```suggestion
			sanitizeIndex(obj.pick(cleanedIndex, identicalProperties) as IndexItem), 
			sanitizeIndex(obj.pick(searchIndex as any, identicalProperties) as IndexItem)
```

**Review by fishcharlie — APPROVED** [review]

**Comment by fishcharlie** [comment]
@madhusudhand Sorry for the delay on this. Thank you for continuing to politely pester me.

**Pull request merged**